### PR TITLE
Fix initialization of `StorageURLSource`

### LINKED ISSUE #29670: `url` table function does not process globs in parallel
```
SELECT * FROM url('https://hacker-news.firebaseio.com/v0/item/{1..100}.json', JSONEachRow,
$$
id UInt32,
deleted UInt8,
type Enum('story' = 1, 'comment' = 2, 'poll' = 3, 'pollopt' = 4, 'job' = 5),
by LowCardinality(String),
time DateTime,
text String,
dead UInt8,
parent UInt32,
poll UInt32,
kids Array(UInt32),
url String,
score Int32,
title String,
parts Array(UInt32),
descendants Int32
$$)
```

All requests are done from single thread.

### LINKED ISSUE #29671: `EXPLAIN` does not work as expected for a query with `url` table function
It performs unneeded requests:

```
EXPLAIN SELECT * FROM url('https://hacker-news.firebaseio.com/v0/item/{1..100}.json', JSONEachRow,　$$　id UInt32,　deleted UInt8,　type Enum('story' = 1, 'comment' = 2, 'poll' = 3, 'pollopt' = 4, 'job' = 5),　by LowCardinality(String),　time DateTime,　text String,　dead UInt8,　parent UInt32,　poll UInt32,　kids Array(UInt32),　url String,　score Int32,　title String,　parts Array(UInt32),　descendants Int32　$$)
```

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Make `url` table function to process multiple URLs in parallel. This closes #29670 and closes #29671.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
`SELECT * FROM url('http://127.0.0.{1..10}:8123/?query=SELECT+sleep(1)', TSV, 'x UInt8')`

**Comment by alexey-milovidov** [comment]
I did not implement it for URL with failover, maybe @kssenii can help.

**Review by kitaisreal — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Integration tests have failed for reason.

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
Integration tests ok, other failures not related.

File: src/Storages/StorageURL.cpp [hunk 1]
```diff
@@ -71,56 +97,62 @@ namespace
...
-            // Propagate OpenTelemetry trace context, if any, downstream.
-            if (CurrentThread::isInitialized())
+            auto headers = getHeaders(headers_);
+            /// Lazy initialization. We should not perform requests in constructor, because we need to do it in query pipeline.
+            initialize = [=, this]
             {
-                const auto & thread_trace_context = CurrentThread::get().thread_trace_context;
-                if (thread_trace_context.trace_id != UUID())
+                WriteBufferFromOwnString error_message;
+                for (auto option = uri_options.begin(); option < uri_options.end(); ++option)
```
Comment (alexey-milovidov) [inline src/Storages/StorageURL.cpp:110]: Shouldn't be `const auto &`?

**Pull request merged**